packages/grid/src/DataBarCellRenderer.ts

```diff
@@ -103,166 +137,99 @@ class DataBarCellRenderer extends CellRenderer {
       dataBarWidth,
     } = this.getDataBarRenderMetrics(context, state, column, row);
 
-    if (this.heightOfDigits === undefined) {
-      const { actualBoundingBoxAscent, actualBoundingBoxDescent } =
-        context.measureText('1234567890');
-      this.heightOfDigits = actualBoundingBoxAscent + actualBoundingBoxDescent;
-    }
-
     context.save();
```

Comment:
With the levels of context.saves(), it'd be nice if they were at the start/end of a block, extracting this info out or just enclosing it in blocks. Maybe even have our own function for it?

```typescript
function withContextState(context, callback: () => undefined) {
  context.save();
  callback();
  context.restore();
}
```

Just musing though. Not necessary to change, but I did find it a little confusing seeing two `context.restore()`s close to each other below, and then tracing it back to the matching `context.save()`

Reply:
Ya that would be neat and I agree it is a bit confusing. It's because you have to clip first and then draw, so 

I added the 2nd context save/restore because I needed it for re-using the gradient across multiple parts of the grid (or even multiple grids since I made the cache static on the renderer)

I think I might be able to simplify some of it and at least put both restores near each other by using the flip/translate for everything based on LTR or RTL. Basically we don't need to calculate special positioning based on if it's LTR or RTL, but instead calculate just based on LTR. Then if it's RTL, the context `scale(-1, 1)` w/ translation makes it draw RTL even though the coordinates are LTR. I already did this w/ the gradient, but could also add it for the markers and axis